shopinvader/services/cart.py

```diff
@@ -292,7 +292,7 @@ def _check_call_onchange(self, params):
     def _confirm_cart(self, cart):
         cart.action_confirm_cart()
         res = self._to_json(cart)
-        res.update(
+        self.shopinvader_session_data.update(
```

Comment:
I'd use a method instead, like: `update_shopinvader_session(**kw)`.
Not a blocker but I think is nicer and eases overrides.

Reply:
Maybe invader_session ?
in a long terme we can have project without shop feature so as this will one day move into an abstract module it's better to use the futur name

Reply:
you mean "update_invader_session"? Then I'd say we should just call it `update_client_session` as this is what we are actually doing, no?

Reply:
@sebastienbeau @simahawk For sure the current implementation is very basic and we will have to improve it in the future. IMO, to do it right, the session (it's more a shopinvader context than a session) should be an object that could be extended in the various addons as needed and providing a clean API. There is still a lot of work to be done to clean up the different places where these data are handled. But that's not really the purpose of this PR (It would be a very good working subject for the OCA days).
The goal of this PR is to provide an entry point to be able to put in place a clean api for shopinvader_payment. (https://github.com/acsone/odoo-shopinvader-payment/pull/5/files)